typescript/cli/src/deploy/warp.ts

```diff
@@ -388,16 +383,25 @@ export async function runWarpRouteApply(
       context.registry,
     );
 
-  const { multiProtocolSigner } = context;
-  assert(multiProtocolSigner, `multiProtocolSigner not defined`);
-
   // temporarily configure deployer as owner so that warp update after extension
   // can leverage JSON RPC submitter on new chains
   const intermediateOwnerConfig = await promiseObjAll(
-    objMap(params.warpDeployConfig, async (chain, config) => ({
-      ...config,
-      owner: await multiProtocolSigner.getSignerAddress(chain),
-    })),
+    objMap(params.warpDeployConfig, async (chain, config) => {
+      switch (context.multiProvider.getProtocol(chain)) {
+        case ProtocolType.Ethereum: {
+          return {
+            ...config,
+            owner: await context.multiProvider.getSignerAddress(chain),
+          };
+        }
+        default: {
+          return {
+            ...config,
+            owner: context.altVmSigner.get(chain).getSignerAddress(),
+          };
+        }
```

Comment:
this will crash the deployment if we have unsupported chains with foreign deployment in the deploy config. After merging https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/7156 this should hopefully be reported in the e2e tests too

Reply:
This issue was unfortunately existing before and this PR does not intend to fix it. Looking forward to your PR getting merged fixing this issue